### DISCUSSION TITLE
Why is x-init  called twice?

### DISCUSSION THREAD

**Discussion by romansh** [livewire/livewire#6018] [1. Help] [answered]
Test.php
```php
<?php

namespace App\Livewire;
use Livewire\Component;

class Test extends Component
{

    public function render()
    {
        return view('livewire.test');
    }
}
```

test.blade.php

```php

<div
    x-data
    x-init="
        () => {
            console.log('Init')  
        }
    "
>
</div>
```
I can see in the console 'Init'  showing up twice, but I expect to see it only once.

**Comment by PhiloNL** [comment]
Move the function to x-data https://alpinejs.dev/directives/init#auto-evaluate-init-method

**Comment by romansh** [comment]
test.blade.php
```php
<div x-data="{
    init() {
        console.log('I am called automatically')
    }
}">
    ...
</div>
``` 
VM13721:5 I am called automatically
VM13733:5 I am called automatically

Edit: removed uBlock dependent message.

**Reply by PhiloNL** [reply, marked as answer]
Any chance you are loading Alpine twice? Livewire itself now includes it in v3.

**Comment by romansh** [comment]
That was it! 

I had 

```js
import Alpine from 'alpinejs';
import focus from '@alpinejs/focus';
window.Alpine = Alpine;

Alpine.plugin(focus);

Alpine.start();
```

in resources/js/app.js

Thank you!
